linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_files_permissions/bash/shared.sh

```diff
@@ -14,9 +14,17 @@ declare -a LOG_FILE_PATHS
 declare -a RSYSLOG_CONFIGS
 RSYSLOG_CONFIGS+=("${RSYSLOG_ETC_CONFIG}" "${RSYSLOG_INCLUDE_CONFIG[@]}" "${RSYSLOG_INCLUDE[@]}")
 
-# Browse each file selected above as containing paths of log files
+# Get full list of files to be checked
 # ('/etc/rsyslog.conf' and '/etc/rsyslog.d/*.conf' in the default configuration)
-for LOG_FILE in "${RSYSLOG_CONFIGS[@]}"
+declare -a RSYSLOG_FILES
+for ENTRY in "${RSYSLOG_CONFIGS[@]}"
+do
+     mapfile -t FINDOUT < <(find "$(dirname ${ENTRY}\")" -maxdepth 1 -name "$(basename \"${ENTRY})")
+     RSYSLOG_FILES+=("${RSYSLOG_FILES[@]}" "${FINDOUT[@]}")
```

Comment:
I don't know why the \" are needed, but they are; code does not work without them. I have tested the result on SLE15 and Ubuntu 20.04 and it works as expected on both. And passes spell check.

Reply:
Code updated so comment above no longer applies.